kernel-spark/src/test/java/io/delta/kernel/spark/read/SparkMicroBatchStreamGetStartingVersionTest.java

```diff
@@ -0,0 +1,297 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.spark.read;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import io.delta.kernel.spark.SparkDsv2TestBase;
+import java.io.File;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.sql.delta.DeltaLog;
+import org.apache.spark.sql.delta.DeltaOptions;
+import org.apache.spark.sql.delta.sources.DeltaSQLConf;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import scala.Option;
+import scala.collection.immutable.Map$;
+
+/** Tests for SparkMicroBatchStream.getStartingVersion parity between DSv1 and DSv2. */
+public class SparkMicroBatchStreamGetStartingVersionTest extends SparkDsv2TestBase {
+
+  /**
+   * Parameterized test that verifies parity between DSv1 DeltaSource.getStartingVersion and DSv2
+   * SparkMicroBatchStream.getStartingVersion.
+   */
+  @ParameterizedTest
+  @MethodSource("getStartingVersionParameters")
+  public void testGetStartingVersion(
+      String startingVersion, Optional<Long> expectedVersion, @TempDir File tempDir)
+      throws Exception {
+    String testTablePath = tempDir.getAbsolutePath();
+    String testTableName = "test_starting_version_" + System.nanoTime();
+    createEmptyTestTable(testTablePath, testTableName);
+
+    // Create 5 versions (version 0 = CREATE TABLE, versions 1-5 = INSERTs)
+    createVersions(testTableName, 5);
+
+    testAndCompareStartingVersion(
+        testTablePath, startingVersion, expectedVersion, "startingVersion=" + startingVersion);
+  }
+
+  /** Provides test parameters for the parameterized getStartingVersion test. */
+  private static Stream<Arguments> getStartingVersionParameters() {
+    return Stream.of(
+        Arguments.of(/* startingVersion= */ "0", /* expectedVersion= */ Optional.of(0L)),
+        Arguments.of(/* startingVersion= */ "1", /* expectedVersion= */ Optional.of(1L)),
+        Arguments.of(/* startingVersion= */ "3", /* expectedVersion= */ Optional.of(3L)),
+        Arguments.of(/* startingVersion= */ "5", /* expectedVersion= */ Optional.of(5L)),
+        Arguments.of(/* startingVersion= */ "latest", /* expectedVersion= */ Optional.of(6L)),
+        Arguments.of(/* startingVersion= */ null, /* expectedVersion= */ Optional.empty()));
```

Comment:
What about these cases:
1. startingVersion is not set
2.  startingVersion set to Optional.empty